testng-core/src/main/java/org/testng/TestClass.java

```diff
@@ -16,7 +16,7 @@
  * This class represents a test class: - The test methods - The configuration methods (test and
  * method) - The class file
  */
-class TestClass extends NoOpTestClass implements ITestClass, ITestClassConfigInfo {
+class TestClass extends NoOpTestClass implements ITestClass, ITestClassConfigInfo, IObject {
```

Comment:
For the record: this class should be deprecated and moved to the internal package.

Reply:
Will do that.